### PR TITLE
Repl enhancement

### PR DESCRIPTION
**Pull request by glennblock**
Adds @migueldeicaza's PR and some cleanup / fixing of tests.

### CODE REVIEW

File: src/ScriptCs.Hosting/FileConsole.cs [hunk 1]
```diff
@@ -35,9 +35,9 @@ public void WriteLine(string value)
             this.AppendLine(value);
         }
 
-        public string ReadLine()
+        public string ReadLine(string prompt)
```
Comment (filipw) [inline src/ScriptCs.Hosting/FileConsole.cs]: why is there `prompt` if it's not being used?

Comment (glennblock) [inline src/ScriptCs.Hosting/FileConsole.cs]: It is used.

Comment (filipw) [inline src/ScriptCs.Hosting/FileConsole.cs]: the whole method is

```
        public string ReadLine(string prompt)
        {
            var line = _innerConsole.ReadLine("");
            this.AppendLine(line);
            return line;
        }
```

Comment (khellang) [inline src/ScriptCs.Hosting/FileConsole.cs]: It's just not used in the `FileConsole`.

Comment (filipw) [inline src/ScriptCs.Hosting/FileConsole.cs]: ah now i see, ok!

Comment (glennblock) [inline src/ScriptCs.Hosting/FileConsole.cs]: Correct, it is part of the general interface

**Comment by filipw** [comment]
ok let's merge this, if something is still discovered to be wrong please make a corrective PR

I wanna play with this

**Pull request merged**